File: Data Structures & Algorithms/missing-number/submission-1.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        m = 0
        counter = 0
        hel = 0
        ans = 0
        for n in nums:
            if n > m:
                m = n
            ans ^= counter
            hel ^= n
            counter += 1
        ans ^= counter
        return ans ^ hel
        
            

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number}
     */
    subarraySum(nums: number[], k: number): number {
        let c = 0;
        let s = [0];
        for (const n of nums) {
            c += n;
            s.push(c);
        }
        let ans = 0;
        for (let i = 0; i <= nums.length; i++) {
            for (let j = i + 1; j <= nums.length; j++) {
                if (s[j] - s[i] === k) {
                    ans += 1;
                }
            }
        }
        return ans;
    }
}
